src/event.rs

```diff
@@ -86,7 +95,7 @@ pub enum Event {
     /// Our own section has been split, resulting in the included `Prefix` for our new section.
     SectionSplit(Prefix<XorName>),
     /// The client has successfully connected to a proxy node on the network.
-    Connected,
+    Connected(ConnectEvent),
```

Comment:
Minor: I might be cleaner to have a separate `Event::Relocated` event.

Reply:
Could be, Can probably be a separate PR if that make sense.
The thing is that these events represent the same state of node (reaching adulthood), so I'm not 100% clear that would be better.

Reply:
When relocating, the node does not really reach adulthood, as it was already adult (or elder). So we might even chose to not raise any event at all. But maybe it is useful for vault? 

I don't mind postponing this to a separate PR though.

Reply:
I would think that having been an Adult in a previous section does not help very much being adult in a new section, if there is stuff you need to do, probably need to be done again in new section. However, yes, lets leave it for later as it does not really impact the rest of this PR.